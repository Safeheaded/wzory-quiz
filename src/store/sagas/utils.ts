import { firestore } from 'firebase';
import { RootReducer } from '../types/main';

export function collectionToArray<T extends K & { id: string }, K>(
    snapshot: firestore.QuerySnapshot
): T[] {
    const items: T[] = [];
    snapshot.forEach(item => {
        const newItem = { id: item.id, ...(item.data() as K) } as T;
        items.push(newItem);
    });
    return items;
}

export const getTopics = (state: RootReducer) => state.topicsReducer.topics;
