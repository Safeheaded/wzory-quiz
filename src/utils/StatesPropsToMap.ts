import { EquationsState } from '../store/types/Equations';

export const mapEqState = ({ eqReducer }: { eqReducer: EquationsState }) => ({
    subjects: eqReducer.subjects,
    topics: eqReducer.topics,
    equations: eqReducer.equations
});
