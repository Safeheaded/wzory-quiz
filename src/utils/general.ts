import process from 'process';
export function sleep(ms: number) {
    return new Promise(resolve => setTimeout(resolve, ms));
}

export default function isDev(): boolean {
    const development: boolean =
        !process.env.NODE_ENV || process.env.NODE_ENV === 'development';
    return development;
}

export function getParams<T extends object>(params: object): T {
    return params as T;
}
