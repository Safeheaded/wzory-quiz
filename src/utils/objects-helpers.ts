export function updateValues<T extends object, K extends T>(
    origin: T,
    source: K
): T {
    let updatedOrigin = {} as T;
    for (const field in origin) {
        if (origin.hasOwnProperty(field) && source.hasOwnProperty(field)) {
            updatedOrigin[field] = source[field];
        }
    }
    return updatedOrigin;
}
