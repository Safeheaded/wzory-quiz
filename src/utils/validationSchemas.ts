import * as Yup from 'yup';

export type LoginValues = { email: string; password: string };

export type EditValues = {
    equation: string;
    name: string;
    subjectRef: string;
    topicRef: string;
    explanations: string[];
};

export function loginSchema() {
    return Yup.object().shape({
        email: Yup.string()
            .required()
            .email(),
        password: Yup.string().required()
    });
}

export function editSchema() {
    return Yup.object().shape({
        equation: Yup.string().required(),
        name: Yup.string().required(),
        subjectRef: Yup.string().required(),
        topicRef: Yup.string().required(),
        explanations: Yup.array().of(Yup.string().required())
    });
}
