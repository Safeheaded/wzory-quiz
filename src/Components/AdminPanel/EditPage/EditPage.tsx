import React, { Component, Fragment, ComponentState } from 'react';
import MathInput from '../MathInput/MathInput';
import { MenuItem } from '@material-ui/core';
import { Dispatch } from 'redux';
import {
    AddEquation,
    ExtendedEquation,
    ExtendedEquationWithId,
    FetchEquation,
    UpdateEquation,
    DeleteEquation
} from '../../../store/types/Equations';
import {
    fetchEquation,
    updateEquation,
    addEquation,
    deleteEquation
} from '../../../store/actions/Equations';
import { connect } from 'react-redux';
import styles from './MainPage.module.sass';
import AddIcon from '@material-ui/icons/Add';
import FormInput from '../FormInput/FormInput';
import FormSelect from '../FormSelect/FormSelect';
import { onChangeType, WriteMode } from '../../../types/admin';
import AddDialog from '../AddDialog/AddDialog';
import { withRouter, RouteComponentProps } from 'react-router-dom';
import { mapEqState, EqStateProps } from '../../../utils/StatesPropsToMap';
import FormActions from './FormActions/FormActions';
import {
    SubjectWithId,
    FetchAllSubjects,
    Subject,
    AddSubject
} from '../../../store/types/Subjects';
import {
    ExtendedTopicWithId,
    FetchAllTopics,
    ExtendedTopic,
    AddTopic
} from '../../../store/types/Topics';
import { fetchAllSubjects, addSubject } from '../../../store/actions/Subjects';
import { fetchAllTopics, addTopic } from '../../../store/actions/Topics';

export interface Props extends RouteComponentProps, EqStateProps {
    url: string;
    addEquation: (equation: ExtendedEquation) => AddEquation;
    fetchAllSubjects: () => FetchAllSubjects;
    addSubject: (subject: Subject) => AddSubject;
    fetchAllTopics: (subjectRef: string) => FetchAllTopics;
    addTopic: (topic: ExtendedTopic) => AddTopic;
    fetchEquation: (id: string) => FetchEquation;
    updateEquation: (equation: ExtendedEquationWithId) => UpdateEquation;
    deleteEquation: (id: string) => DeleteEquation;
}

type params = { id: string };

export interface State {
    subjectDialogState: boolean;
    topicDialogState: boolean;
    subjectRef: string;
    topicRef: string;
    equationId?: string;
    mode: WriteMode;
    explanation: string;
    equation: string;
}

type inputTypes = HTMLSelectElement | HTMLInputElement | HTMLTextAreaElement;

export class EditPage extends Component<Props, State> {
    state: State = {
        subjectDialogState: false,
        subjectRef: '',
        topicRef: '',
        topicDialogState: false,
        mode: WriteMode.Add,
        equation: '',
        explanation: ''
    };

    componentDidMount() {
        this.props.fetchAllSubjects();
        const equationId = (this.props.match.params as params).id;
        if (equationId) {
            this.setState({
                mode: WriteMode.Edit,
                equationId
            });
            this.extractEquation(equationId);
        }
    }

    private extractEquation(equationId: string) {
        if (this.props.equations.length === 0) {
            this.props.fetchEquation(equationId);
        } else {
            this.getEquation(equationId);
        }
    }

    private getEquation(equationId: string) {
        const equation = this.findEquation(equationId);
        this.handleEquationData(equation);
    }

    private handleEquationData(equation: ExtendedEquationWithId | undefined) {
        if (equation) {
            this.props.fetchAllTopics(equation.subjectRef);
            this.setState({
                subjectRef: equation.subjectRef,
                explanation: equation.explanation,
                equation: equation.equation
            });
        }
    }

    private findEquation(equationId: string) {
        return this.props.equations.find(
            equation => equation.id === equationId
        );
    }

    componentDidUpdate(prevProps: Props, prevState: State) {
        const equationFetched = prevProps.equations !== this.props.equations;
        if (equationFetched && prevState.equationId) {
            this.getEquation(prevState.equationId);
        }
        if (prevProps.topics !== this.props.topics) {
            this.setTopic();
        }
    }

    onSubmitHandler = (event: React.FormEvent<HTMLFormElement>) => {
        event.preventDefault();
        const data = new FormData(event.target as HTMLFormElement);
        const equation: ExtendedEquation = {
            equation: data.get('equation') as string,
            explanation: data.get('explanation') as string,
            subjectRef: data.get('subjectRef') as string,
            topicRef: data.get('topicRef') as string
        };
        if (this.state.mode === WriteMode.Add) {
            this.props.addEquation(equation);
        } else {
            this.props.updateEquation({
                ...equation,
                id: this.state.equationId as string
            });
        }
    };

    addSubjectHandler = () => {
        this.setState({ subjectDialogState: true });
    };

    onSelectChange = (event: onChangeType, lastItemValue?: string) => {
        const value = (event.target as inputTypes).value;
        const target = (event.target as inputTypes).name as keyof State;
        if (lastItemValue) {
            this.setValueOrOpenDialog(value, lastItemValue, target);
        } else {
            this.setState({ [target]: value } as ComponentState);
        }
    };

    onAddSubject = (event: React.FormEvent<HTMLFormElement>) => {
        event.preventDefault();
        const data = new FormData(event.target as HTMLFormElement);
        const subject: Subject = { name: data.get('name') as string };
        this.props.addSubject(subject);
    };

    onAddTopic = (event: React.FormEvent<HTMLFormElement>) => {
        event.preventDefault();
        const data = new FormData(event.target as HTMLFormElement);
        const topic: ExtendedTopic = {
            name: data.get('name') as string,
            subjectRef: this.state.subjectRef
        };
        this.props.addTopic(topic);
    };

    private setTopic() {
        const equation = this.props.equations.find(
            equation => equation.id === this.state.equationId
        );
        if (equation) {
            this.setState({ topicRef: equation.topicRef });
        }
    }

    private setValueOrOpenDialog(
        value: string,
        lastItemValue: string | undefined,
        fieldToUpdate: keyof State
    ) {
        const dialog =
            lastItemValue === 'add_subject'
                ? 'subjectDialogState'
                : 'topicDialogState';
        if (value === lastItemValue) {
            this.setState({ [dialog]: true } as ComponentState);
        } else {
            this.setSubjectOrTopic(fieldToUpdate, value);
        }
    }

    private setSubjectOrTopic(fieldToUpdate: string, value: string) {
        if (fieldToUpdate === 'subjectRef') {
            this.updateSubject(value);
        } else {
            this.setState({ topicRef: value });
        }
    }

    private updateSubject(value: string) {
        const subject: SubjectWithId | undefined = this.props.subjects.find(
            subject => subject.id === value
        );
        if (subject) {
            this.props.fetchAllTopics(subject.id);
        }
        this.setState({ subjectRef: value });
    }

    toggleDialog = (field: string) => {
        const fieldToUpdate = field as keyof State;
        this.setState(
            prevState =>
                ({
                    [fieldToUpdate]: !prevState[fieldToUpdate]
                } as ComponentState)
        );
    };

    deleteEquationHandler = () => {
        this.props.deleteEquation(this.state.equationId as string);
        this.props.history.replace('/admin');
    };

    render() {
        const subjectLastItem = (
            <MenuItem value="add_subject">
                Dodaj przedmiot <AddIcon />
            </MenuItem>
        );

        const topicLastItem = (
            <MenuItem value="add_topic">
                Dodaj temat <AddIcon />
            </MenuItem>
        );
        return (
            <Fragment>
                <form
                    className={styles.Form}
                    onSubmit={e => this.onSubmitHandler(e)}
                >
                    <MathInput
                        value={this.state.equation}
                        onValueChange={(e: onChangeType) =>
                            this.onSelectChange(e)
                        }
                    />
                    <FormInput
                        label="Znaczenie rÃ³wnania"
                        rows="5"
                        fullWidth
                        multiline
                        name="explanation"
                        onValueChange={(e: onChangeType) =>
                            this.onSelectChange(e)
                        }
                        value={this.state.explanation}
                    />
                    <FormSelect
                        value={this.state.subjectRef}
                        name="subjectRef"
                        id="subject"
                        lastItem={subjectLastItem}
                        label="Przedmioty"
                        onValueChange={(e: onChangeType) =>
                            this.onSelectChange(e, 'add_subject')
                        }
                        values={this.props.subjects}
                    />

                    <FormSelect
                        value={this.state.topicRef}
                        name="topicRef"
                        id="topic"
                        lastItem={topicLastItem}
                        label="Tematy"
                        onValueChange={(e: onChangeType) =>
                            this.onSelectChange(e, 'add_topic')
                        }
                        values={this.props.topics}
                        disabled={this.state.subjectRef === '' ? true : false}
                    />

                    <FormActions
                        secondaryButtonAction={this.deleteEquationHandler}
                        mode={this.state.mode}
                    />
                </form>

                <AddDialog
                    label="Przedmiot"
                    title="Dodaj przedmiot"
                    toggleDialog={() => this.toggleDialog('subjectDialogState')}
                    stateField={this.state.subjectDialogState}
                    onSubmit={this.onAddSubject}
                />

                <AddDialog
                    label="Temat"
                    title="Dodaj temat"
                    toggleDialog={() => this.toggleDialog('topicDialogState')}
                    stateField={this.state.topicDialogState}
                    onSubmit={this.onAddTopic}
                />
            </Fragment>
        );
    }
}

const mapDispatchToProps = (dispatch: Dispatch) => ({
    addEquation: (equation: ExtendedEquation) =>
        dispatch(addEquation(equation)),
    fetchAllSubjects: () => dispatch(fetchAllSubjects()),
    addSubject: (subject: Subject) => dispatch(addSubject(subject)),
    fetchAllTopics: (subjectRef: string) =>
        dispatch(fetchAllTopics(subjectRef)),
    addTopic: (topic: ExtendedTopic) => dispatch(addTopic(topic)),
    fetchEquation: (id: string) => dispatch(fetchEquation(id)),
    updateEquation: (equation: ExtendedEquationWithId) =>
        dispatch(updateEquation(equation)),
    deleteEquation: (id: string) => dispatch(deleteEquation(id))
});

export default withRouter(connect(mapEqState, mapDispatchToProps)(EditPage));
