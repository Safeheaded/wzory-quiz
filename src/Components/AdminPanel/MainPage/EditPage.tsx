import React, { Component, Fragment, ComponentState } from 'react';
import MathInput from '../MathInput/MathInput';
import { Button, MenuItem, FormControl } from '@material-ui/core';
import { Dispatch } from 'redux';
import {
    AddEquationActionType,
    ExtendedEquation,
    FetchAllSubjectsActionType,
    SubjectWithId,
    Subject,
    AddSubjectActionType,
    ExtendedTopicWithId,
    FetchAllTopicsActionType,
    ExtendedTopic,
    AddTopicActionType,
    ExtendedEquationWithId,
    FetchEquationActionType
} from '../../../store/types/Equations';
import {
    addEquation,
    fetchAllSubjects,
    addSubject,
    fetchAllTopics,
    addTopic,
    fetchEquation
} from '../../../store/actions/Equations';
import { connect } from 'react-redux';
import styles from './MainPage.module.sass';
import AddIcon from '@material-ui/icons/Add';
import FormInput from '../FormInput/FormInput';
import FormSelect from '../FormSelect/FormSelect';
import { onChangeType } from '../../../types/admin';
import { RootReducer } from '../../../store/types/main';
import AddDialog from '../AddDialog/AddDialog';
import { withRouter, RouteComponentProps } from 'react-router-dom';
import { mapEqState } from '../../../utils/StatesPropsToMap';

interface Props extends RouteComponentProps {
    equations: ExtendedEquationWithId[];
    subjects: SubjectWithId[];
    topics: ExtendedTopicWithId[];
    url: string;
    addEquation: (equation: ExtendedEquation) => AddEquationActionType;
    fetchAllSubjects: () => FetchAllSubjectsActionType;
    addSubject: (subject: Subject) => AddSubjectActionType;
    fetchAllTopics: (subjectRef: string) => FetchAllTopicsActionType;
    addTopic: (topic: ExtendedTopic) => AddTopicActionType;
    fetchEquation: (id: string) => FetchEquationActionType;
}

type params = { id: string };

interface State {
    subjectDialogState: boolean;
    topicDialogState: boolean;
    subjectValue: string;
    topicValue: string;
    equationId?: string;
    mode: Mode;
    explanation: string;
    equation: string;
}

enum Mode {
    Edit,
    Add
}

type inputTypes = HTMLSelectElement | HTMLInputElement | HTMLTextAreaElement;

class MainPage extends Component<Props, State> {
    state: State = {
        subjectDialogState: false,
        subjectValue: '',
        topicValue: '',
        topicDialogState: false,
        mode: Mode.Add,
        equation: '',
        explanation: ''
    };

    componentDidMount() {
        this.props.fetchAllSubjects();
        const equationId = (this.props.match.params as params).id;
        if (equationId) {
            this.setState({
                mode: Mode.Edit,
                equationId
            });
            this.extractEquation(equationId);
        }
    }

    private extractEquation(equationId: string) {
        if (this.props.equations.length === 0) {
            this.props.fetchEquation(equationId);
        } else {
            this.getEquation(equationId);
        }
    }

    private getEquation(equationId: string) {
        const equation = this.findEquation(equationId);
        this.handleEquationData(equation);
    }

    private handleEquationData(equation: ExtendedEquationWithId | undefined) {
        if (equation) {
            this.props.fetchAllTopics(equation.subjectRef);
            this.setState({
                subjectValue: equation.subjectRef,
                explanation: equation.explanation,
                equation: equation.equation
            });
        }
    }

    private findEquation(equationId: string) {
        return this.props.equations.find(
            equation => equation.id === equationId
        );
    }

    componentDidUpdate(prevProps: Props, prevState: State) {
        const equationFetched = prevProps.equations !== this.props.equations;
        if (equationFetched && prevState.equationId) {
            this.getEquation(prevState.equationId);
        }
        if (prevProps.topics !== this.props.topics) {
            this.setTopic();
        }
    }

    onSubmitHandler = (event: React.FormEvent<HTMLFormElement>) => {
        event.preventDefault();
        const data = new FormData(event.target as HTMLFormElement);
        const equation: ExtendedEquation = {
            equation: data.get('equation') as string,
            explanation: data.get('explanation') as string,
            subjectRef: data.get('subjectRef') as string,
            topicRef: data.get('topicRef') as string
        };
        this.props.addEquation(equation);
    };

    addSubjectHandler = () => {
        this.setState({ subjectDialogState: true });
    };

    onSelectChange = (event: onChangeType, lastItemValue?: string) => {
        const value = (event.target as inputTypes).value;
        const target = (event.target as inputTypes).name as keyof State;
        if (lastItemValue) {
            this.setValueOrOpenDialog(value, lastItemValue, target);
        } else {
            this.setState({ [target]: value } as ComponentState);
        }
    };

    onAddSubject = (event: React.FormEvent<HTMLFormElement>) => {
        event.preventDefault();
        const data = new FormData(event.target as HTMLFormElement);
        const subject: Subject = { name: data.get('name') as string };
        this.props.addSubject(subject);
    };

    onAddTopic = (event: React.FormEvent<HTMLFormElement>) => {
        event.preventDefault();
        const data = new FormData(event.target as HTMLFormElement);
        const topic: ExtendedTopic = {
            name: data.get('name') as string,
            subjectRef: this.state.subjectValue
        };
        this.props.addTopic(topic);
    };

    private setTopic() {
        const equation = this.props.equations.find(
            equation => equation.id === this.state.equationId
        );
        if (equation) {
            this.setState({ topicValue: equation.topicRef });
        }
    }

    private setValueOrOpenDialog(
        value: string,
        lastItemAction: string | undefined,
        fieldToUpdate: keyof State
    ) {
        if (value === lastItemAction) {
            this.setState({ [fieldToUpdate]: true } as ComponentState);
        } else {
            this.setSubjectOrTopic(fieldToUpdate, value);
        }
    }

    private setSubjectOrTopic(fieldToUpdate: string, value: string) {
        if (fieldToUpdate === 'subjectDialogState') {
            this.updateSubject(value);
        } else {
            this.updateTopic(value);
        }
    }

    private updateTopic(value: string) {
        if (value === 'add_topic') {
            this.setState({ topicDialogState: true });
        } else {
            this.setState({ topicValue: value });
        }
    }

    private updateSubject(value: string) {
        const subject: SubjectWithId | undefined = this.props.subjects.find(
            subject => subject.id === value
        );
        if (subject) {
            this.props.fetchAllTopics(subject.id);
        }
        this.setState({ subjectValue: value });
    }

    toggleDialog = (field: string) => {
        const fieldToUpdate = field as keyof State;
        this.setState(
            prevState =>
                ({
                    [fieldToUpdate]: !prevState[fieldToUpdate]
                } as ComponentState)
        );
    };

    render() {
        const subjectLastItem = (
            <MenuItem value="add_subject">
                Dodaj przedmiot <AddIcon />
            </MenuItem>
        );

        const topicLastItem = (
            <MenuItem value="add_topic">
                Dodaj temat <AddIcon />
            </MenuItem>
        );
        return (
            <Fragment>
                <form
                    className={styles.Form}
                    onSubmit={e => this.onSubmitHandler(e)}
                >
                    <MathInput
                        value={this.state.equation}
                        onValueChange={(e: onChangeType) =>
                            this.onSelectChange(e)
                        }
                    />
                    <FormInput
                        label="Znaczenie równania"
                        rows="5"
                        fullWidth
                        multiline
                        name="explanation"
                        onValueChange={(e: onChangeType) =>
                            this.onSelectChange(e)
                        }
                        value={this.state.explanation}
                    />
                    <FormSelect
                        value={this.state.subjectValue}
                        name="subjectRef"
                        id="subject"
                        lastItem={subjectLastItem}
                        label="Przedmioty"
                        onValueChange={(e: onChangeType) =>
                            this.onSelectChange(e, 'add_subject')
                        }
                        values={this.props.subjects}
                    />

                    <FormSelect
                        value={this.state.topicValue}
                        name="topicRef"
                        id="topic"
                        lastItem={topicLastItem}
                        label="Tematy"
                        onValueChange={(e: onChangeType) =>
                            this.onSelectChange(e, 'add_topic')
                        }
                        values={this.props.topics}
                        disabled={this.state.subjectValue === '' ? true : false}
                    />

                    <FormControl>
                        <Button
                            variant="contained"
                            color="primary"
                            type="submit"
                        >
                            Wyślij
                        </Button>
                    </FormControl>
                </form>

                <AddDialog
                    label="Przedmiot"
                    title="Dodaj przedmiot"
                    toggleDialog={() => this.toggleDialog('subjectDialogState')}
                    stateField={this.state.subjectDialogState}
                    onSubmit={this.onAddSubject}
                />

                <AddDialog
                    label="Temat"
                    title="Dodaj temat"
                    toggleDialog={() => this.toggleDialog('topicDialogState')}
                    stateField={this.state.topicDialogState}
                    onSubmit={this.onAddTopic}
                />
            </Fragment>
        );
    }
}

const mapDispatchToProps = (dispatch: Dispatch) => ({
    addEquation: (equation: ExtendedEquation) =>
        dispatch(addEquation(equation)),
    fetchAllSubjects: () => dispatch(fetchAllSubjects()),
    addSubject: (subject: Subject) => dispatch(addSubject(subject)),
    fetchAllTopics: (subjectRef: string) =>
        dispatch(fetchAllTopics(subjectRef)),
    addTopic: (topic: ExtendedTopic) => dispatch(addTopic(topic)),
    fetchEquation: (id: string) => dispatch(fetchEquation(id))
});

export default withRouter(connect(mapEqState, mapDispatchToProps)(MainPage));
