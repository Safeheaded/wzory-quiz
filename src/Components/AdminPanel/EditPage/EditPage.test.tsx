import React from 'react';
import { shallow } from 'enzyme';
import { EditPage, Props } from './EditPage';
import { getRouterTestProps } from '../../../tests/utils/testRouteProps';

const routeProps = getRouterTestProps('/admin/add-equation', {});
const props = {
    ...routeProps,
    url: routeProps.match.url,
    addEquation: jest.fn(),
    addSubject: jest.fn(),
    addTopic: jest.fn(),
    deleteEquation: jest.fn(),
    fetchAllSubjects: jest.fn(),
    fetchAllTopics: jest.fn(),
    fetchEquation: jest.fn(),
    updateEquation: jest.fn(),
    subjects: [],
    equations: [],
    topics: []
} as Props;

describe('<EditPage /> general tests', () => {
    it('Should match snapshot', () => {
        const wrapper = shallow(<EditPage {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
});
