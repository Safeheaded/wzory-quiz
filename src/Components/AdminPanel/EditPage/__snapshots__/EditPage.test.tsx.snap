// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditPage /> general tests Should match snapshot 1`] = `
<Fragment>
  <form
    className="Form"
    onSubmit={[Function]}
  >
    <MathInput
      onValueChange={[Function]}
      value=""
    />
    <FormInput
      fullWidth={true}
      label="Znaczenie rÃ³wnania"
      multiline={true}
      name="explanation"
      onValueChange={[Function]}
      rows="5"
      value=""
    />
    <FormSelect
      id="subject"
      label="Przedmioty"
      lastItem={
        <ForwardRef(WithStyles)
          value="add_subject"
        >
          Dodaj przedmiot 
          <UNDEFINED />
        </ForwardRef(WithStyles)>
      }
      name="subjectRef"
      onValueChange={[Function]}
      value=""
      values={Array []}
    />
    <FormSelect
      disabled={true}
      id="topic"
      label="Tematy"
      lastItem={
        <ForwardRef(WithStyles)
          value="add_topic"
        >
          Dodaj temat 
          <UNDEFINED />
        </ForwardRef(WithStyles)>
      }
      name="topicRef"
      onValueChange={[Function]}
      value=""
      values={Array []}
    />
    <FormActions
      mode={1}
      secondaryButtonAction={[Function]}
    />
  </form>
  <AddDialog
    label="Przedmiot"
    onSubmit={[Function]}
    stateField={false}
    title="Dodaj przedmiot"
    toggleDialog={[Function]}
  />
  <AddDialog
    label="Temat"
    onSubmit={[Function]}
    stateField={false}
    title="Dodaj temat"
    toggleDialog={[Function]}
  />
</Fragment>
`;
